src/containers/shared/components/Transaction/SetHook/utils.ts

```diff
@@ -0,0 +1,134 @@
+import { zeroPad } from '../../../transactionUtils'
+
+// TODO: import from ripple-binary-codec
+const TRANSACTION_TYPES: Record<string, number> = {
+  Invalid: -1,
+  Payment: 0,
+  EscrowCreate: 1,
+  EscrowFinish: 2,
+  AccountSet: 3,
+  EscrowCancel: 4,
+  SetRegularKey: 5,
+  NickNameSet: 6,
+  OfferCreate: 7,
+  OfferCancel: 8,
+  Contract: 9,
+  TicketCreate: 10,
+  TicketCancel: 11,
+  SignerListSet: 12,
+  PaymentChannelCreate: 13,
+  PaymentChannelFund: 14,
+  PaymentChannelClaim: 15,
+  CheckCreate: 16,
+  CheckCash: 17,
+  CheckCancel: 18,
+  DepositPreauth: 19,
+  TrustSet: 20,
+  AccountDelete: 21,
+  SetHook: 22,
+  NFTokenMint: 25,
+  NFTokenBurn: 26,
+  NFTokenCreateOffer: 27,
+  NFTokenCancelOffer: 28,
+  NFTokenAcceptOffer: 29,
+  Invoke: 99,
+  EnableAmendment: 100,
+  SetFee: 101,
+  UNLModify: 102,
+  EmitFailure: 103,
+}
+
+const transactionMap = Object.entries(TRANSACTION_TYPES).reduce(
+  (flipped, [key, value]) => {
+    // eslint-disable-next-line no-param-reassign -- fine for a reduce
+    flipped[value] = key
+    return flipped
+  },
+  {} as Record<number, string>,
+)
+
+const maxTransactionValue: number = 103
+
+function hex2bin(input) {
```

Comment:
is this function from somewhere already? 
there could be a simplification using `parseInt` and `padStart`
something like: 
`
function hex2bin(input) {
    let hex = input.replace('0x', '').toLowerCase();
    let bin = '';
    for (let i = 0; i < hex.length; i++) {
        let binFragment = parseInt(hex[i], 16).toString(2);
        bin += binFragment.padStart(4, '0');
    }
    return bin;
}`

Reply:
The other `hex2bin` didn't work, because the number in decimal would be too large. I'll try this, though